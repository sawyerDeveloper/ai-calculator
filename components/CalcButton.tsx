import {
  StyleSheet,
  Text,
  Pressable,
  ColorValue,
} from 'react-native';

interface CalcButtonProps {
  value: number | string;
  label: string;
  action: (calcValue: number | string) => void;
  backgroundColor?: ColorValue;
}
export function CalcButton({
  value,
  label,
  action,
  backgroundColor,
}: CalcButtonProps) {
  return (
    <Pressable
      onPress={() => action(value)}
      style={[styles.container, { backgroundColor: backgroundColor }]}
    >
      <Text>{label}</Text>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 50,
    height: 50,
    borderColor: 'black',
    borderWidth: 1,
    cursor: 'pointer',
  },
});
