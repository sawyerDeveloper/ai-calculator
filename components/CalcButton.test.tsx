import renderer from 'react-test-renderer';
import { CalcButton } from './CalcButton';

it('renders correctly', () => {
  const tree = renderer
    .create(<CalcButton value="a" label="a" action={() => {}} backgroundColor="000000" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});