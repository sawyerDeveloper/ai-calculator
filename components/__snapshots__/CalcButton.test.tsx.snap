// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "black",
        "borderWidth": 1,
        "cursor": "pointer",
        "fontSize": 20,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      },
      {
        "backgroundColor": "000000",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontWeight": "600",
        },
        {
          "fontSize": 30,
        },
      ]
    }
  >
    a
  </Text>
</View>
`;
